src/test.js

```diff
@@ -338,10 +336,9 @@ Test.prototype = {
 		if ( promise != null ) {
 			then = promise.then;
 			if ( QUnit.objectType( then ) === "function" ) {
-				QUnit.stop();
-				then.call(
-					promise,
-					function() { QUnit.start(); },
+				internalStop( test );
+				promise.then(
```

Comment:
Believe it or not, it's important to keep the `then.call`: https://promisesaplus.com/#point-75